presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/TestMergeAdjacentWindows.java

```diff
@@ -0,0 +1,174 @@
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner.iterative.rule.test;
+
+import com.facebook.presto.metadata.FunctionKind;
+import com.facebook.presto.metadata.Signature;
+import com.facebook.presto.sql.planner.assertions.ExpectedValueProvider;
+import com.facebook.presto.sql.planner.iterative.rule.MergeAdjacentWindows;
+import com.facebook.presto.sql.planner.plan.Assignments;
+import com.facebook.presto.sql.planner.plan.WindowNode;
+import com.facebook.presto.sql.tree.FunctionCall;
+import com.facebook.presto.sql.tree.QualifiedName;
+import com.facebook.presto.sql.tree.SymbolReference;
+import com.facebook.presto.sql.tree.Window;
+import com.facebook.presto.sql.tree.WindowFrame;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.testng.annotations.Test;
+
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.spi.type.DoubleType.DOUBLE;
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.functionCall;
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.specification;
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.values;
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.window;
+import static com.facebook.presto.sql.tree.FrameBound.Type.CURRENT_ROW;
+import static com.facebook.presto.sql.tree.FrameBound.Type.UNBOUNDED_PRECEDING;
+
+public class TestMergeAdjacentWindows
+{
+    private final RuleTester tester = new RuleTester();
+
+    private WindowNode.Frame frame;
+    private Signature signature;
+
+    public TestMergeAdjacentWindows()
+    {
+        frame = new WindowNode.Frame(WindowFrame.Type.RANGE, UNBOUNDED_PRECEDING,
+                Optional.empty(), CURRENT_ROW, Optional.empty());
+        signature = new Signature(
+                "avg",
+                FunctionKind.WINDOW,
+                ImmutableList.of(),
+                ImmutableList.of(),
+                DOUBLE.getTypeSignature(),
+                ImmutableList.of(DOUBLE.getTypeSignature()),
+                false);
+    }
+
+    @Test
+    public void testDoesNotFireOnPlanWithoutWindowFunctions()
+            throws Exception
+    {
+        tester.assertThat(new MergeAdjacentWindows())
+                .on(p -> p.values(p.symbol("a", BIGINT)))
+                .doesNotFire();
+    }
+
+    @Test
+    public void testDoesNotFireOnPlanWithSingleWindowNode()
+            throws Exception
+    {
+        tester.assertThat(new MergeAdjacentWindows())
+                .on(p -> p.window(
+                        newWindowNodeSpecification(p, "a"),
+                        ImmutableMap.of(p.symbol("avg_1", BIGINT), newWindowNodeFunction("avg", "a")),
+                        p.values(p.symbol("a", BIGINT))))
```

Comment:
it looks like indentation here is broken, please make `p.window` to be in separated line as in test below.